packages/kbn-esql-validation-autocomplete/src/autocomplete/autocomplete.test.ts

```diff
@@ -286,42 +275,53 @@ describe('autocomplete', () => {
-      const constantPattern = command === 'grok' ? '"%{WORD:firstWord}"' : '"%{firstWord}"';
-      const subExpressions = [
-        '',
-        `${command} stringField |`,
-        `${command} stringField ${constantPattern} |`,
-        `dissect stringField ${constantPattern} append_separator = ":" |`,
-      ];
-      if (command === 'grok') {
-        subExpressions.push(`dissect stringField ${constantPattern} |`);
-      }
-      for (const subExpression of subExpressions) {
-        testSuggestions(`from a | ${subExpression} ${command} `, getFieldNamesByType('string'));
-        testSuggestions(`from a | ${subExpression} ${command} stringField `, [constantPattern]);
-        testSuggestions(
-          `from a | ${subExpression} ${command} stringField ${constantPattern} `,
-          (command === 'dissect' ? ['APPEND_SEPARATOR = $0'] : []).concat(['|'])
-        );
-        if (command === 'dissect') {
-          testSuggestions(
-            `from a | ${subExpression} ${command} stringField ${constantPattern} append_separator = `,
-            ['":"', '";"']
-          );
-          testSuggestions(
-            `from a | ${subExpression} ${command} stringField ${constantPattern} append_separator = ":" `,
-            ['|']
-          );
-        }
-      }
-    });
-  }
+  describe('grok', () => {
+    const constantPattern = '"%{WORD:firstWord}"';
+    const subExpressions = [
+      '',
+      `grok stringField |`,
+      `grok stringField ${constantPattern} |`,
+      `dissect stringField ${constantPattern} append_separator = ":" |`,
+      `dissect stringField ${constantPattern} |`,
+    ];
+    for (const subExpression of subExpressions) {
+      testSuggestions(`from a | ${subExpression} grok `, getFieldNamesByType('string'));
+      testSuggestions(`from a | ${subExpression} grok stringField `, [constantPattern], ' ');
+      testSuggestions(`from a | ${subExpression} grok stringField ${constantPattern} `, ['|']);
+    }
+  });
+
+  describe('dissect', () => {
+    const constantPattern = '"%{firstWord}"';
+    const subExpressions = [
+      '',
+      `dissect stringField |`,
+      `dissect stringField ${constantPattern} |`,
+      `dissect stringField ${constantPattern} append_separator = ":" |`,
+    ];
+    for (const subExpression of subExpressions) {
+      testSuggestions(`from a | ${subExpression} dissect `, getFieldNamesByType('string'));
+      testSuggestions(`from a | ${subExpression} dissect stringField `, [constantPattern], ' ');
+      testSuggestions(
+        `from a | ${subExpression} dissect stringField ${constantPattern} `,
+        ['APPEND_SEPARATOR = $0', '|'],
+        ' '
+      );
+      testSuggestions(
+        `from a | ${subExpression} dissect stringField ${constantPattern} append_separator = `,
+        ['":"', '";"']
+      );
+      testSuggestions(
+        `from a | ${subExpression} dissect stringField ${constantPattern} append_separator = ":" `,
+        ['|']
+      );
+    }
+  });
```

Comment:
This expansion was unrelated to the main point of the PR, but made the tests much easier to work with.